iib/workers/tasks/build.py

```diff
@@ -826,6 +837,11 @@ def handle_add_request(
             'index.Dockerfile',
         )
 
+        if is_fbc:
+            index_db_file = os.path.join(temp_dir, get_worker_config()['temp_index_db_path'])
+            # make sure FBC is generated right before build
+            opm_migrate(index_db=index_db_file, base_dir=temp_dir)
```

Comment:
So we are migrating inside ` opm_registry_add_fbc` and here. May I ask why? Shouldn't we only do it in one place?

Reply:
Sorry I just noticed this

Reply:
My intention was to have the same functionality in `opm_registry_add_fbc` as we do have in `opm_index_add`.
That is why I have `opm_migrate` and `opm_generate_dockerfile` inside of `opm_registry_add_fbc`.

The issue is that in `handle_add_request` we call also `add_max_ocp_version_property` which is modifying index.db by `_add_property_to_index`. Therefore we have to run migration again. To be sure that file-based catalog is up to date with any changes.

I was considering to move it to build_image. But then I did not do it because it would mean generating file-based catalog again for all specified arches. And it seemed redundant to me (doing it multiple times without any change).

Reply:
ack, makes sense. Thanks @lipoja